src/provider/sequencer.ts

```diff
@@ -74,9 +75,13 @@ export class SequencerProvider implements ProviderInterface {
       this.gatewayUrl = urljoin(this.baseUrl, 'gateway');
     } else {
       this.baseUrl = optionsOrProvider.baseUrl;
-      this.feederGatewayUrl =
-        optionsOrProvider.feederGatewayUrl ?? urljoin(this.baseUrl, 'feeder_gateway');
-      this.gatewayUrl = optionsOrProvider.gatewayUrl ?? urljoin(this.baseUrl, 'gateway');
+      this.feederGatewayUrl = this.parseUrlOrPath(
+        this.baseUrl,
+        'feeder_gateway',
+        optionsOrProvider.feederGatewayUrl
+      );
+      this.gatewayUrl = this.parseUrlOrPath(this.baseUrl, 'gateway', optionsOrProvider.gatewayUrl);
```

Comment:
Nitpick about style difference between feederGatewayUrl and gatewayUrl (new lines vs one line)

Reply:
My prettier config might have done that

Reply:
EDIT: I think this one is okay as it comes from `printWidth: 100` defined in [.prettierrc]( https://github.com/0xs34n/starknet.js/blob/08ec9c6215b153f3bcb098fcc641f1ba3669d3e7/.prettierrc#L2)